pkg/controller/v1beta1/inferenceservice/utils/utils.go

```diff
@@ -441,3 +443,19 @@ func MergeServingRuntimeAndInferenceServiceSpecs(srContainers []v1.Container, is
 	}
 	return containerIndexInSR, mergedContainer, mergedPodSpec, nil
 }
+
+func RemoveCookieSecretArg(deployment appsv1.Deployment) *appsv1.Deployment {
```

Comment:
Since the cookie secret is generated randomly, during the reconcile loop when the existing vs desired deployment is compared, the comparison results in a `update` every loop because the cookie-secret arg is different every time it is generated. We want to exclude the random string argument while comparing existing and desired deployments.

Reply:
sounds good, please add it as comment in the code base.